DTAConfig/Settings/FileSourceDestinationInfo.cs

```diff
@@ -151,6 +134,35 @@ public void Revert()
                         $"Invalid {nameof(FileOperationOptions)} value of {FileOperationOptions}");
             }
         }
+
+        [DllImport("Kernel32.dll", SetLastError = true, CharSet = CharSet.Unicode)]
+        private static extern bool CreateHardLink(string lpFileName, string lpExistingFileName, IntPtr lpSecurityAttributes);
+
+        /// <summary>
+        ///  Creates a link from the source file to the destination file. 
+        ///  Source and destination paths must be relative because for
+        ///  .NET 8 in Linux method creates symbolic link with relative paths.
+        /// </summary>
+        private void CreateLinkToFile(string source, string destination)
+        {
+#if NETFRAMEWORK
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                bool result = CreateHardLink(destination, source, IntPtr.Zero);
+                if (!result)
+                    throw new Exception(Marshal.GetExceptionForHR(Marshal.GetHRForLastWin32Error()).Message);
+            }
```

Comment:
missing else

Reply:
if you mean add `File.CreateSymbolicLink` -- it isn't possible due to .NET 4.8 limitations. `File.CreateSymbolicLink` needs .NET 6 or newer.
Reference: https://learn.microsoft.com/en-us/dotnet/api/system.io.file.createsymboliclink?view=net-8.0

```cs
else
    File.CreateSymbolicLink(destination, source); // this will get compilation error in .NET 4.8
```

Reply:
throw an Exception

Reply:
> throw an Exception

`bool result` contains the result of working `CreateHardLink`. And `!result` triggers when it is a `false` value.

Reply:
No. Currently, your method will not throw an exception for .NET 4.8 on non-Windows platform.

Reply:
> No. Currently, your method will not throw an exception for .NET 4.8 on non-Windows platform.

Maybe better remove OS check for .NET 4.8? For Windows it would work as always, and WINE on Linux could wrap it.

Reply:
There is also a mono case. 

Don't remove if-windows here. Throw an exception. The reasons were listed earlier

Reply:
> There is also a mono case.
> 
> Don't remove if-windows here. It's better to explicitly throw an exception. The reasons were listed earlier

Ok, what if for Linux i would be use `link` from `libc.so`?

https://www.man7.org/linux/man-pages/man2/link.2.html

```cs
else
{
    int result = link(destination, source);
    if (result != 0)
        throw new Exception(string.Format("Unable to create hard link to file {0}", source));
}
```

Reply:
> > There is also a mono case.
> > Don't remove if-windows here. It's better to explicitly throw an exception. The reasons were listed earlier
> 
> Ok, what if for Linux i would be use `link` from `libc.so`?
> 
> https://www.man7.org/linux/man-pages/man2/link.2.html
> 
> ```cs
> else
> {
>     int result = link(source, destination);
>     if (result != 0)
>         throw new Exception(string.Format("Unable to create hard link to file {0}", source));
> }
> ```

I won't recommend so since you will have to dynamically link both glibc (with a minimum version number) and musl-libc for a wide Linux support. This causes a lot of efforts in .NET.